### ISSUE TITLE
Authorization Header issue

### ISSUE THREAD

**Issue by danway2012** [AxaFrance/react-oidc#1056]
I have made an Authorization Server with Spring Security that requires to send an Authorization Header with the Bearer token. To get the token and access token Authentication Header must include client_id and client_secret, if not returns a 401 status. Is there anyway to include Authentication Header in the POST method to  /oauth2/token endpoint?

Thanks for the help :)

**Comment by danway2012** [comment]
I could fix the issue, adding to RegisteredClientRepository this: .clientAuthenticationMethod(ClientAuthenticationMethod.CLIENT_SECRET_POST) Thanks to this, Spring Security could manage client_id and client_secret in x-www-form-urlencoded

**Comment by danway2012** [comment]
Firstly, I want to thank you for this amazing library and for your promptly reply. I have been searching for a while and this is one of the best implementation of Ouath2 client with OpenId. While, I was having this issue, I check out the documentation for  The OAuth 2.0 Authorization Framework: Bearer Token Usage. This is what I found in rfc6749:

![image](https://github.com/AxaFrance/react-oidc/assets/80462934/208ca021-8fc0-4d74-8c7e-d9b62c87e00a)

https://www.rfc-editor.org/rfc/rfc6750 (This is the reference where I found the rfc 6749, mentioned in the introduction section)

For example when Postman send POST Request to /oauth2/token, includes de Authorization Header to include clientId and client Secret. Show you an example:

![image](https://github.com/AxaFrance/react-oidc/assets/80462934/2722aa7c-03e1-47a8-b37a-b513d445b6f0)

If you decode the Authorization Header:
![image](https://github.com/AxaFrance/react-oidc/assets/80462934/7239623e-fd82-4caa-b1cb-9bf263df2f71)
Format -> clientId:clientSecret
Basic is just for the telling to Authorization Server that Client Authentication Method is CLIENT_SECRET_BASIC.

My question about custom Login Page was if there is anyway to connect to the Login endpoint of Authorization Server to send a POST with de fields needed but to not complicate myself anymore, I will just change Login Authorization Server page.

Hope this help you to implement this new feature :)
